client/src/components/Cell/CellPets.tsx

```diff
@@ -20,6 +34,16 @@ const CellPets = ({
     setHoveredPet(null);
   };
 
+  const onPetClick = useCallback((clicked: PetName) => () => {
```

Comment:
No need for a useCallback here, as it's a double function, and the second one isn't memoized this way.